gef.py

```diff
@@ -689,10 +689,33 @@ def size(self) -> int:
             raise AttributeError
         return self.page_end - self.page_start
 
+    def search_for_realpath(self) -> str:
+        """This function is a workaround for gdb bug #23764
+
+        path might be wrong for remote sessions, so try a simple search for files
+        that aren't found at the path indicated, which should be canonical.
+        """
+
+        remote_path = pathlib.Path(self.path)
+        # First, try the canonical path in the remote session root.
+        candidate1 = gef.session.remote.root / remote_path.relative_to(remote_path.anchor)
```

Comment:
you need to check if the session is remote before using `gef.session.remote` otherwise, it will throw an exception

Reply:
Right now, search_for_realpath() is only called after checking if the session is remote (line 718).  Would you like an assert here?

Reply:
(Or just mark the function private, so that it doesn't gain new users which haven't already checked gef.session.remote?)

Reply:
An assert will probably let us detect bugs and give better errors if we ever hit that situation.

Reply:
You could also make it so that you pass in `gef.remote` as an argument.

Reply:
I've added an assert.

Passing gef.session.remote as an argument to a private function doesn't make much sense to me, since the realpath property that calls the search function doesn't take arguments.

Reply:
FTR I'm very happy with the `assert`, like graz said "if" it fails we'll know immediately why